### PR TITLE
adding failurereason and failuremessage

### PR DESCRIPTION
**Pull request by yannickstruyf3**
**What this PR does / why we need it**:
- Adds Failurereason and Failuremessage attributes to the NutanixCluster and NutanixMachine specs
- Failure messages are added in case of errors.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A

**Special notes for your reviewer**:
Steps performed to test:
- Define incorrect input parameters for cluster creation. For example:
     - Incorrect image
     - Incorrect subnet
     - Incorrect PE
- Verify the NutanixMachine CR and check the status spec
  
**Release note**:

```release-note
- Adds Failurereason and Failuremessage attributes to the NutanixCluster and NutanixMachine specs
```

### CODE REVIEW

**Review by tuxtof — DISMISSED** [review]

File: controllers/nutanixcluster_controller.go [hunk 1]
```diff
@@ -181,6 +182,10 @@ func (r *NutanixClusterReconciler) reconcileDelete(rctx *nctx.ClusterContext) (r
 
 func (r *NutanixClusterReconciler) reconcileNormal(rctx *nctx.ClusterContext) (reconcile.Result, error) {
 
+	if rctx.NutanixCluster.Status.FailureReason != nil || rctx.NutanixCluster.Status.FailureMessage != nil {
```
Comment (deepakm-ntnx) [inline controllers/nutanixcluster_controller.go:185]: what is the way to re-trigger reconciliation in this case then?

Comment (deepakm-ntnx) [reply controllers/nutanixcluster_controller.go:185]: also what about errors during delete reconciliation?

Comment (yannickstruyf3) [reply controllers/nutanixcluster_controller.go:185]: FailureReason and FailureMessage are only set when a Fatal error occurs hence we stop the loop in this case.

**Comment by tuxtof** [comment]
@yannickstruyf3 can you rebase your PR please

**Comment by yannickstruyf3** [comment]
Rebased

**Review by tuxtof — APPROVED** [review]

**Review by deepakm-ntnx — APPROVED** [review]

**Pull request merged**